EMMItem.cs

```diff
@@ -47,6 +119,14 @@ public override GlobalItem Clone(Item item, Item itemClone)
 		public override void Load(Item item, TagCompound tag)
 		{
 			GetItemInfo(item).Modifier = Modifier._Load(tag);
+
+			// Apply on load. DUH.
+			ModifierContext ctx = new ModifierContext
```

Comment:
What you're doing here is great, however, I don't think it's great applying the effects with the OnPickup context. It might be better if we'd store the passed contexts when they were being applied, which could be saved and loaded in _Load and _Save, and then could be passed here. We could still use ApplyItem, the difference being we'd have the contexts stored when they were being applied to the item.

Reply:
I understand the bit about sending OnPickup, though I could simply not set a context method at all (only sending Item). About the other part... I don't get it. ApplyItem only runs once: either when the item gets modifiers or the item loads, and for good reason, as running these effects multiple times would only distort the item's current stats beyond what the effects should be. What effect would care about whether the item was rolled or loading?

When an item rerolls (once that's implemented) it will have to SetDefaults to get its stats back to normal, then use ApplyItem again.

Reply:
Modders may influence behaviour of modifiers to their own desire based on present context at that time. It is a use case that must be considered, therefore we should track them as noted in my previous comment. Just in regards to not understanding the other 'part': they're not parts. It's one text and meant as one thing. We shouldn't pass any other context than ones we had stored (i.e. we should pass the stored context, rather than OnPickup) Just as a note, we're not actually storing any contexts just yet, just so you're not getting confused as to where this code may be: we should make it.

Reply:
I still don't get it. At all. We're gonna need to find some time to talk about this, as I am still under the belief that we may as well remove "context" from ApplyItem completely.